ethereum/contracts/relayer/coreRelayer/CoreRelayerDelivery.sol

```diff
@@ -326,56 +345,75 @@ abstract contract CoreRelayerDelivery is CoreRelayerGovernance {
     function payRefundRemote(
         IWormholeRelayerInternalStructs.DeliveryInstruction memory instruction,
         uint256 refundAmount,
-        IRelayProvider provider
-    ) internal returns(bool, RefundStatus){
-        IWormhole wormhole = wormhole();
-        uint256 wormholeMessageFee = wormhole.messageFee();
-        uint256 overhead = wormholeMessageFee + provider.quoteDeliveryOverhead(instruction.refundChain);
+        address providerAddress
+    ) internal returns (bool, RefundStatus) {
+        bool isChainSupported;
+        uint256 wormholeMessageFee = wormhole().messageFee();
+        uint256 receiverValueTarget;
+        uint256 maximumBudget;
+        address rewardAddress;
+
+        if (refundAmount <= wormholeMessageFee) {
+            return (false, RefundStatus.CROSS_CHAIN_REFUND_FAIL_NOT_ENOUGH);
+        }
 
-        if (refundAmount > overhead) {
-            (IWormholeRelayerInternalStructs.DeliveryInstruction memory refundInstruction, bool isMaximumBudget) = getInstructionForEmptyMessageWithReceiverValue(
-                        instruction.refundChain, instruction.refundAddress, refundAmount - overhead, provider
-            );
-            if(refundInstruction.receiverValueTarget == 0) {
-                return (false, RefundStatus.CROSS_CHAIN_REFUND_FAIL_NOT_ENOUGH);
-            }
-            wormhole.publishMessage{value: wormholeMessageFee}(
-                0, encodeDeliveryInstruction(refundInstruction), refundInstruction.consistencyLevel
-            );
+        (isChainSupported, rewardAddress, maximumBudget, receiverValueTarget) =
+            getValuesFromRelayProvider(providerAddress, instruction.refundChain, refundAmount - wormholeMessageFee);
 
-            pay(provider.getRewardAddress(), refundAmount - wormholeMessageFee);
+        if (!isChainSupported) {
+            return (false, RefundStatus.CROSS_CHAIN_REFUND_FAIL_PROVIDER_NOT_SUPPORTED);
+        }
 
-            return (true, isMaximumBudget ? RefundStatus.CROSS_CHAIN_REFUND_SENT_MAXIMUM_BUDGET : RefundStatus.CROSS_CHAIN_REFUND_SENT);
-            
-        } else {
+        if (receiverValueTarget == 0) {
```

Comment:
can you put a comment here saying that the overhead was checked in the lib?

Reply:
done